src/plugins/intel_gpu/src/graph/impls/ocl/prior_box.cpp

```diff
@@ -89,7 +89,7 @@ struct prior_box_impl : typed_primitive_impl_ocl<prior_box> {
 namespace detail {
 
 attach_prior_box_impl::attach_prior_box_impl() {
-    auto types = {data_types::i32, data_types::i64};
+    auto types = {data_types::i32, data_types::i64, data_types::f32, data_types::f16};
```

Comment:
Just out of curiosity : why this work of refactoring requires changes in dtypes?

Reply:
Because it's not only refactoring 😄 

This kind of changes is needed since I've changed the meaning of registered precision+layout elements in legacy map. Previously the definition was blurry, and in different contexts precision/layout was checked for either input or output. And in some cases we were just lucky that another port wasn't checked.

Now, the registered elements reflects supported layout/precision for both input and output, and when we check impl availability, we actually check both. That's why priorbox needs fp types support - inputs to the op has fp type.